### PR TITLE
CI for QuickChick and ext-lib

### PR DESCRIPTION
**Pull request by lemonidas**
**Kind:** infrastructure.

### CODE REVIEW

**Review by ejgallego — APPROVED** [review]
LGTM

**Comment by ejgallego** [comment]
Umm, @lemonidas , the QuickChick makefile seems to be returning `1`, can you check what's going on? As we discussed, `make ci-quickchick` on an updated coq build should do the work.

**Comment by Zimmi48** [comment]
If I understand correctly, QuickChick's master branch tracks Coq master and QuickChick's 8.8 branch tracks Coq v8.8. Note that the README is saying that the master branch depends on Coq 8.7 so this should be fixed.

File: dev/ci/ci-quickchick.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+ci_dir="$(dirname "$0")"
+
+# This script could be included inside other ones
+# Let's avoid to source ci-common twice in this case
+if [ -z "${CI_BUILD_DIR}" ];
```
Comment (SkySkimmer) [inline dev/ci/ci-quickchick.sh]: Don't bother with this conditional.

File: dev/ci/ci-ext-lib.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+ci_dir="$(dirname "$0")"
+
+# This script could be included inside other ones
+# Let's avoid to source ci-common twice in this case
+if [ -z "${CI_BUILD_DIR}" ];
```
Comment (SkySkimmer) [inline dev/ci/ci-ext-lib.sh]: Don't bother with this conditional.

File: dev/ci/ci-quickchick.sh (hunk 1 above)
Comment (ejgallego) [reply dev/ci/ci-quickchick.sh]: Note that this is done already in some files, hence the origin from the cut-and-paste.

**Comment by ejgallego** [comment]
Needs to be rebased on top of #7680

**Comment by ejgallego** [comment]
This is still broken, strange, as the log doesn't seem very informative. Are you sure your makefile is safe in the use of `make -j`. It is easy to write `Makefiles` not safe when enabling a paralell build.

**Comment by SkySkimmer** [comment]
QuickChick is using the record-in-constructor ocaml feature which isn't available in 4.02.3:
https://github.com/QuickChick/QuickChick/blob/c513d3178f0fad8c8351438fce97083b283a3ab8/quickChickTool/quickChickToolTypes.ml#L15-L27

File: dev/ci/ci-quickchick.sh (hunk 1 above)
Comment (SkySkimmer) [reply dev/ci/ci-quickchick.sh]: >Note that this is done already in some files, hence the origin from the cut-and-paste.

That's no reason to keep the copy when flagged in review though.

**Comment by ejgallego** [comment]
Oh, thanks @SkySkimmer , this explains why I couldn't reproduce. [looking forward to having build-envs for the CI]

@lemonidas , can you then use the `ci-template-flambda` template in the gitlab file?

[We could also add a `ci-template-edge` I guess]

File: dev/ci/ci-quickchick.sh (hunk 1 above)
Comment (lemonidas) [reply dev/ci/ci-quickchick.sh]: Do you want me to remove the entire conditional altogether or just remove the condition and leave the "then" branch body?

**Pull request merged**